repose-aggregator/components/filters/herp-filter/src/test/scala/org/openrepose/filters/herp/HerpFilterTest.scala

```diff
@@ -355,10 +392,60 @@ class HerpFilterTest extends FunSpec with BeforeAndAfterEach with Matchers {
       herpFilter.doFilter(servletRequest, servletResponse, filterChain)
 
       // then:
-      def logEvents = listAppenderPre.getEvents
+      val logEvents = listAppenderPre.getEvents
       logEvents.size shouldBe 1
       logEvents.get(0).getMessage.getFormattedMessage should include("\"DefaultProjectID\" : \"bar\"")
     }
+    it("should extract and log the default request tenant id header from Response if not available from Request") {
+      // given:
+      addResponseHeaders(List(("X-Tenant-Id", "foo;q=0.5"), ("X-Tenant-Id", "bar;q=1.0")))
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      val logEvents = listAppenderPre.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"DefaultProjectID\" : \"bar\"")
+    }
+
+    val hdrNames = List("X-Tenant-Id", "X-Project-Id")
+    val reqValues = List(List("reqFoo;q=0.5", "reqBar;q=1.0"), List("reqFoo;q=0.5,reqBar;q=1.0"))
+    val resValues = List(List("resFoo;q=0.5", "resBar;q=1.0"), List("resFoo;q=0.5,resBar;q=1.0"))
+    val headers = for {
+      reqHdr <- hdrNames
+      resHdr <- hdrNames
+    } yield (reqHdr, resHdr)
```

Comment:
I to am surprised there is no easier way to do this, but I have used your other way before too. Again I don't have a strong opinion on the single line vs the multiline and both are equally readable.
Fixed.